main.tf

```diff
@@ -139,7 +138,7 @@ module "aks" {
   aks_cluster_os_disk_size                 = var.default_nodepool_os_disk_size
   aks_cluster_node_vm_size                 = var.default_nodepool_vm_type
   aks_cluster_node_admin                   = var.node_vm_admin
-  aks_cluster_ssh_public_key               = file(var.ssh_public_key)
+  aks_cluster_ssh_public_key               = try( file(var.ssh_public_key), "")
```

Comment:
Need to verify this will work if the file does not populate the value. This input variable is required in the module so having it set to "" or even null may be an issue.

Reply:
I did change the module code to deal with an empty value. Pls have a look at the change in modules/azure_aks/main.tf  in this PR.